src/pytorch_lightning/core/module.py

```diff
@@ -270,6 +271,11 @@ def loggers(self) -> List[Logger]:
         """Reference to the list of loggers in the Trainer."""
         return self.trainer.loggers if self._trainer else []
 
+    @property
+    def compiler(self) -> Optional[str]:
```

Comment:
I would have imagined a slightly different approach.

Either introduce a new hook:

```python
from lightning.pytorch import Compiler

def configure_compiler(self):
     return Compiler(backend="inductor", nopython=True, dynamic=True)
```

By doing so, if the users override this new hook, we would directly compile the model for them at the right level.

If a user is using DDP, I would expect the entire training_step to be wrapped including the forward method of the DDP wrapper.

or 

```python
Trainer(

   compile=Compiler(backend="inductor", nopython=True, dynamic=True)
)
```

Reply:
As discussed, I'm going to change the design to something different altogether so that we can just have users rely on `torch.compile`